odxtools/templates/macros/printDOP.xml.jinja2

```diff
@@ -12,7 +12,7 @@
 
 {%- macro printDiagCodedType(dct) -%}
 <DIAG-CODED-TYPE {{-make_xml_attrib("BASE-DATA-TYPE", dct.base_data_type.value)}}
-                 {{-make_xml_attrib("BASE-TYPE-ENCODING", dct.base_type_encoding)}}
+                 {{-make_xml_attrib("BASE-TYPE-ENCODING", dct.base_type_encoding and dct.base_type_encoding.value)}}
```

Comment:
Why?

Reply:
because before this PR, `.base_type_encoding` was of type `Optional[str]`, now it is `Optional[Encoding]`.